### ISSUE TITLE
support for multi-dimensional arrays

### ISSUE THREAD

**Issue by jpaas** [danthorpe/ValueCoding#9]
If I have a ValueCoding type, Foo contained inside a multi-dimensional array, e.g.
```swift
var foos = [[Foo]]()
let data = NSKeyedArchiver.archivedDataWithRootObject(foos.encoded)
```
I get an error that says "value of type '[[Foo]]' has no member 'ValueType'".

Is there any way of making this work for a multi-dimensional array?

**Comment by danthorpe** [comment]
Hi @jpaas - that's an interesting use case! I'll have a think about how to do that - PRs are welcome too though if you have an idea.

**Comment by jpaas** [comment]
Sorry @danthorpe I barely understand how this thing works ;) I think my workaround for now has to be to create another ValueCoding type that wraps my 2nd level arrays.

**Comment by danthorpe** [comment]
@jpaas no worries - I think it will be pretty trivial to support one level of nesting. Give me a couple of hours and I should be able to get it sorted.

**Comment by danthorpe** [comment]
Hi @jpaas - I've merged #10 into the `development` branch. You could try building your app against this branch, if you're using CocoaPods:

```ruby
pod 'ValueCoding', :git => 'https://github.com/danthorpe/ValueCoding.git', :branch => 'development'
```

or Carthage:

```
github "danthorpe/ValueCoding" "development"
```

Then you should be able to use `.encoded` on your array of arrays, and `Foo.decode()` on your coders. Let me know how it goes! And Happy New Year!

**Comment by jpaas** [comment]
@danthorpe I've been trying this new code out and although I can archive [[Foo]].encoded, I seem to be having trouble unarchiving it. I can unarchive it, but I'm unable to cast it back to a type of [[Foo]]. `NSKeyedUnarchiver.unarchiveObjectWithData(data) as? [[Foo]]` results in nil.

**Comment by danthorpe** [comment]
Are you trying this:

```swift
var foos = [[Foo]]()
let data = NSKeyedArchiver.archivedDataWithRootObject(foos.encoded)
let decoded = Foo.decode(NSKeyedUnarchiver.unarchiveObjectWithData(data as? [[AnyObject]]))
```

**Comment by jpaas** [comment]
Ah, now that I'm using the decode method it works! Thanks @danthorpe!

**Comment by danthorpe** [comment]
Okay, good stuff. Will close this issue now as it's resolved and released as version 1.2.0.
